core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -61,6 +61,62 @@ class LogCleanerManagerTest extends Logging {
     Utils.delete(tmpDir)
   }
 
+  @Test
+  def testGrabFilthiestCompactedLogThrowsException(): Unit = {
+    // the exception should be catched and the partition that caused it marked as uncleanable
+    class LogMock(dir: File, config: LogConfig, logStartOffset: Long, recoveryPoint: Long, scheduler: Scheduler,
+                  brokerTopicStats: BrokerTopicStats, time: Time, maxProducerIdExpirationMs: Int, producerIdExpirationCheckIntervalMs: Int,
+                  topicPartition: TopicPartition, producerStateManager: ProducerStateManager,
+                  logDirFailureChannel: LogDirFailureChannel) extends Log(dir, config, logStartOffset, recoveryPoint,
+      scheduler, brokerTopicStats, time, maxProducerIdExpirationMs, producerIdExpirationCheckIntervalMs,
+      topicPartition, producerStateManager, logDirFailureChannel) {
+
+      // Throw an error in getFirstBatchTimestampForSegments since it is called in grabFilthiestLog()
+      override def getFirstBatchTimestampForSegments(segments: Iterable[LogSegment]): Iterable[Long] =
+        throw new IllegalStateException("Error!")
+    }
+
+    val records = TestUtils.singletonRecords("test".getBytes)
+
+    val tp1 = new TopicPartition("A", 0)
+    val log1: Log = createLog(records.sizeInBytes * 5, LogConfig.Compact, 1, new File(logDir, "A-0"))
+
+    val tp2 = new TopicPartition("A", 1)
+    val log2SegmentSize = records.sizeInBytes * 10
+    val log2SegmentsCount = 2
+    val tp2Dir = new File(logDir, "A-1")
+    val producerStateManager = new ProducerStateManager(tp2, tp2Dir, 60 * 60 * 1000)
+    val log2: Log = new LogMock(tp2Dir, logConfig(log2SegmentSize, LogConfig.Compact),
+      logStartOffset = 0L, recoveryPoint = 0L, scheduler = time.scheduler, time = time,
+      brokerTopicStats = new BrokerTopicStats, maxProducerIdExpirationMs = 60 * 60 * 1000,
+      producerIdExpirationCheckIntervalMs = LogManager.ProducerIdExpirationCheckIntervalMs,
+      topicPartition = topicPartition, producerStateManager = producerStateManager,
+      logDirFailureChannel = new LogDirFailureChannel(10))
+    fillUpLog(log2, log2SegmentsCount)
+
+    val tp3 = new TopicPartition("A", 2)
+    val log3: Log = createLog(records.sizeInBytes * 15, LogConfig.Compact, 3, new File(logDir, "A-2"))
+
+    val logsPool = new Pool[TopicPartition, Log]()
+    logsPool.put(tp1, log1)
+    logsPool.put(tp2, log2)
+    logsPool.put(tp3, log3)
+
+    val cleanerManager: LogCleanerManagerMock = createCleanerManager(logsPool, toMock = true).asInstanceOf[LogCleanerManagerMock]
+    cleanerCheckpoints.put(tp1, 0) // all clean
+    cleanerCheckpoints.put(tp2, 1) // dirtiest - 9 dirty messages
+    cleanerCheckpoints.put(tp3, 15) // 5 dirty messages
+
+    try {
+      cleanerManager.grabFilthiestCompactedLog(time).get
+      fail(s"Expected grabFilthiestCompactedLog call to throw an $ErroneousLogException")
+    } catch {
+      case e: ErroneousLogException =>
```

Comment:
nit: use `assertThrows`

Reply:
I would like to assert the `assertEquals(log, e.log)` check though, is there a way to achieve that with assertThrows?

Reply:
`assertThrows` returns the caught exception.

Reply:
Thanks, I didn't know that. Will use it